### DISCUSSION TITLE
Ansible over teleport not working

### DISCUSSION THREAD

**Discussion by BeoKool** [gravitational/teleport#37731] [Q&A] [answered]
I am trying to use Ansible with Teleport and encountering the current error when running the Ansible command, facing this issue `` fatal: [aht-hrm-02.AHT-VIETTEL]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: \u001b[31mERROR: \u001b[0mfailed connecting to host aht-hrm-02.aht-viettel:22: failed to receive cluster details response\n\tfailed to dial target host\n\tDirect dialing to nodes not found in the inventory is not supported.\nIf you want to connect to a node without installing Teleport on it, consider registering it with\nyour cluster with 'teleport join openssh'.\n\nSee https://goteleport.com/docs/ver/14.x/server-access/guides/openssh/ for more details.\n\nkex_exchange_identification: Connection closed by remote host\r\nConnection closed by UNKNOWN port 65535", "unreachable": true}
`` and I try this command to debug `ssh -vvv -F ./ssh.cfg aht-admin@aht-hrm-02.AHT-VIETTEL` And I encounter the following error: 
```
ERROR: failed connecting to host aht-hrm-02.aht-viettel:22: failed to receive cluster details response
        failed to dial target host
        Direct dialing to nodes not found in the inventory is not supported.
If you want to connect to a node without installing Teleport on it, consider registering it with
your cluster with 'teleport join openssh'.

See https://goteleport.com/docs/ver/14.x/server-access/guides/openssh/ for more details.

kex_exchange_identification: Connection closed by remote host
```

**Comment by stevenGravy** [comment]
So this means it's attempting node names that aren't registered. Are you using OpenSSH servers?   Please see https://goteleport.com/docs/server-access/guides/openssh/openssh/ for registering the nodename and confirm with `tctl nodes ls` what. your nodename is.

**Reply by BeoKool** [reply]
This is the result when I run the command tctl nodes ls --insecure.

![Untitled](https://github.com/gravitational/teleport/assets/83693541/1da9646e-5351-4abc-ad09-eaeebed53c43)

**Reply by stevenGravy** [reply]
Is that node name there?

**Reply by BeoKool** [reply]
yep, is my node name

**Comment by webvictim** [comment, marked as answer]
The issue here is that node names and routing in Teleport are case-sensitive by default.

Your cluster name is `AHT-VIETTEL`, but SSH (and thus Ansible) normalises every hostname to lowercase when trying to connect to it. As such, SSH is actually trying to connect to `aht-hrm-02.aht-viettel`, but the hostname Teleport sees connected is `aht-hrm-02.AHT-VIETTEL` and it treats these as two completely different hostnames.

Try setting this in your `/etc/teleport.yaml` config file on the server which runs your Teleport auth service, then restarting Teleport, and running `ssh -vvv -F ./ssh.cfg aht-admin@aht-hrm-02.AHT-VIETTEL` again:

```yaml
auth_service:
  enabled: true
  case_insensitive_routing: true
```
